### DISCUSSION TITLE
ID3 Series tag

### DISCUSSION THREAD

**Discussion by p-rintz** [advplyr/audiobookshelf#1481] [Q&A] [answered]
Hi there,

Im having an issue getting audiobookshelf to recognize series mp3 tags.

The following ID3 tags of an mp3 as an example, audiobookshelf keeps saying it doesnt have a series.

Is the 'MVNM' tag wrong? Am I doing anything else wrong?

Id appreciate your help. Thank you.

```
APIC=cover front, cover (image/jpeg, 132059 bytes)
MVNM=Die drei ???
TALB=Die drei ??? 172: und der Eisenmann
TBPM=124
TCMP=0
TCON=Kids
TDOR=2014-11-28
TDRC=2016-11-18
TIPL=[unrepresentable data]
TIT2=172 - und der Eisenmann - Teil 01
TLAN=deu
TMED=Digital Media
TPE1=Die drei ???
TPE2=Die drei ???
TPOS=1/1
TPUB=EUROPA
TRCK=1/40
TSOP=drei ???, Die
TSRC=DEC711612738
TXXX=ALBUMARTISTSORT=drei ???, Die
TXXX=Album Artist Credit=Die drei ???
TXXX=Artist Credit=Die drei ???
TXXX=MusicBrainz Album Artist Id=e028fab5-39ae-4ed9-b8c2-c4344d88b171
TXXX=MusicBrainz Album Id=a1140e56-905c-437c-8e65-cd47128eb960
TXXX=MusicBrainz Album Release Country=DE
TXXX=MusicBrainz Album Status=Official
TXXX=MusicBrainz Album Type=o / t / h / e / r / ; /   / a / u / d / i / o /   / d / r / a / m / a
TXXX=MusicBrainz Artist Id=e028fab5-39ae-4ed9-b8c2-c4344d88b171
TXXX=MusicBrainz Release Group Id=57e2af32-8433-4e5c-a66f-17753f7f7cb4
TXXX=MusicBrainz Release Track Id=aded88ad-5ab1-46f3-ae7c-71941d0c1142
TXXX=Script=Latn
UFID=http://musicbrainz.org=b'07478bf7-8aa6-4b0a-be60-82b57e8d1a36'
USLT==XXX=
```

**Comment by advplyr** [comment, marked as answer]
I think @sandreas told me once that the `MVNM` tag isn't picked up by ffprobe.

We are still using ffprobe to scan meta tags at the moment. Just for additional context on that, we included [tone](https://github.com/sandreas/tone) a while back to help with meta tags and is what we are using to write meta tags.

At the moment we still want to use ffprobe when scanning audio files because that will give us info like the audio codec, channels, etc.
We could use both tone and ffprobe in the scanner but I want to make sure that doesn't slow down the scanner much.

**Reply by advplyr** [reply]
Btw you can also put in a custom meta tag `series` that would get picked up. Not ideal but could be the only option atm.

**Reply by p-rintz** [reply]
So it would be something like this?

`TXXX=series=Die drei ???`

**Reply by advplyr** [reply]
I think that should work. I'm not sure how you are embedding meta tags.

**Reply by p-rintz** [reply]
I use mid3v2.

As a sidenote, so that it may help others:

This is wrong:
`mid3v2 --TXXX "series=Insert series name here" /path/to/track`

This is correct:
`mid3v2 --TXXX "series:Insert series name here" /path/to/track`

The first counter-intuitively leads to:
`TXXX==series=Insert series name here`
Instead of:
`TXXX=series=Insert series name here`

Which makes Audiobookshelf not recognize it again.

**Comment by sandreas** [comment]
> I think @sandreas told me once that the MVNM tag isn't picked up by ffprobe.

Yes, unfortunately `ffprobe` does not support `movement`, `movementname` and `movementtotal`.

> At the moment we still want to use ffprobe when scanning audio files because that will give us info like the audio codec, channels, etc.

Since `tone` is a metadata editor and not an encoder or codec analyzer, I do not plan to integrate a detailed stream analysis, so you would have to stick with `ffprobe` for this.

 `ffprobe` is a mighty tool, but be also aware that it may lie to you about the channels as well as the duration. Depending on what info you need, `tone` was more accurate in some of my use cases. Here is an example dump of the same file, where `tone` shows 1 channel (correct) and `ffprobe` shows 2 channels (incorrect):

```bash
 tone dump --format="json" "$AB" --query="$.audio"
```

```json

{
  "bitrate": 64,
  "format": "MPEG-4 Part 14",
  "formatShort": "MPEG-4",
  "sampleRate": 11025.0,
  "duration": 36226008.526,
  "channels": {
    "count": 1,
    "description": "Mono (1/0.0)"
  },
  "frames": {
    "offset": 40,
    "length": 289808076
  },
  "metaFormat": [
    "mp4"
  ]
}
```

```bash
ffprobe -v quiet -print_format json -show_format -show_streams "$AB"
```

```json
{
    "streams": [
        {
            "index": 0,
            "codec_name": "aac",
            "codec_long_name": "AAC (Advanced Audio Coding)",
            "profile": "HE-AAC",
            "codec_type": "audio",
            "codec_time_base": "1/22050",
            "codec_tag_string": "mp4a",
            "codec_tag": "0x6134706d",
            "sample_fmt": "fltp",
            "sample_rate": "22050",
            "channels": 2,
            "channel_layout": "stereo",
            "bits_per_sample": 0,
            "r_frame_rate": "0/0",
            "avg_frame_rate": "0/0",
            "time_base": "1/11025",
            "start_pts": 0,
            "start_time": "0.000000",
            "duration_ts": 399391744,
            "duration": "36226.008526",
            "bit_rate": "63999",
            "max_bit_rate": "60832",
            "nb_frames": "390031"
     }
}
```

**Reply by advplyr** [reply]
Interesting.. I didn't know it could mess up the channels. I have seen issues with duration before but I thought that was just because the metadata was bad.
